torch/onnx/symbolic_helper.py

```diff
@@ -950,29 +963,58 @@ def dequantize_helper(g, qtensor, qdtype=None):
     value = g.op("Cast", tensor, to_i=qdtype)
     scale = g.op("Cast", scale, to_i=torch.onnx.TensorProtoDataType.FLOAT)
     zero_point = g.op("Cast", zero_point, to_i=qdtype)
-    return g.op("DequantizeLinear", value, scale, zero_point), scale, zero_point
 
-def quantize_helper(g, tensor, scale, zero_point):
+    if _export_onnx_opset_version < 13:
+        if axis_i is not None:
+            _onnx_opset_unsupported_detailed(
+                "DequantizeLinear", _export_onnx_opset_version, 13, "Attribute axis is not supported.")
+
+    axis_kwargs = {"axis_i": axis_i} if axis_i is not None else {}
+    return g.op("DequantizeLinear", value, scale, zero_point, **axis_kwargs), scale, zero_point, axis
+
+def quantize_helper(g, tensor, scale, zero_point, axis=None):
+    """Append to graph `g` with ONNX nodes that quantizes `tensor` based on `scale`, `zero_point` and `axis`.
+
+    Args:
+        g: Graph, the ONNX IR graph that is under construction.
+        tensor: torch._C.Value, representing the tensor to be quantized.
+        scale: torch._C.Value, quantized scale.
+        zero_point: torch._C.Value, quantized zero point.
+        axis: Optional[torch._C.Value] default None, if None, represents per tensor quantization.
+          Otherwise, represents per channel quantization, along given axis.
+    """
+    if axis is not None and not _is_none(axis) and _export_onnx_opset_version < 13:
+        _onnx_opset_unsupported_detailed("QuantizeLinear", _export_onnx_opset_version, 13, "Attribute axis is not supported.")
+    axis_kwargs = {"axis_i": _get_const(axis, "i", "axis")}
+
     assert scale is not None
     if scale.type().scalarType() != "Float":
         scale = g.op("Cast", scale, to_i=torch.onnx.TensorProtoDataType.FLOAT)
 
     assert zero_point is not None
     if zero_point.type().scalarType() not in ("Byte", "Char"):
         zero_point = g.op("Cast", zero_point, to_i=torch.onnx.TensorProtoDataType.UINT8)
-    output = g.op("QuantizeLinear", tensor, scale, zero_point)
-    return g.op("prim::TupleConstruct", output, scale, zero_point)
-
-def requantize_bias_helper(g, bias, input_scale, weight_scale):
-    # In PyTorch, bias is float and is quantized implicitly inside the quantized ATen op kernel.
-    # In ONNX we need to make the quantization explicit because operators expect all of their inputs to be quantized.
-    # Since int32 is not supported by ONNX operator `QuantizeLinear`, quantization is exported using regular operators.
+    output = g.op("QuantizeLinear", tensor, scale, zero_point, **axis_kwargs)
```

Comment:
It seems like `axis_kwargs` is set unconditionally. In which case you can get rid of it and just always specify `axis_i`. Or is it a bug that `axis_kwargs` is always set?

```suggestion
    output = g.op("QuantizeLinear", tensor, scale, zero_point, axis_i=_get_const(axis, "i", "axis"))
```

Reply:
Good catch. It turns out for passing attribute, the attribute will not be set if value being passed is None. Hence we can get rid of the kwargs and always specify `axis_i`.